File: elixir/bob/bob.exs
defmodule Bob do
  def hey(input) do
    cond do
      String.last(input) == "?" -> "Sure."
      String.strip(input) == "" -> "Fine. Be that way!"
      String.match?(input, ~r/^[\d\s,]*$/) -> "Whatever."
      String.upcase(input) == input -> "Whoa, chill out!"
      true -> "Whatever."
    end
  end
end

File: ruby/bob/bob.rb
class Bob

  def self.hey(string)
    remark = string.delete("\n")

    case remark
    when remark.upcase == remark && /[A-Z]/
      'Whoa, chill out!'
    when /^.*\?$/
      'Sure.'
    when /^\s*$/
      'Fine. Be that way!'
    else
      'Whatever.'
    end
  end
end
